core/src/link_resolver.rs

```diff
@@ -217,6 +217,58 @@ impl LinkResolverTrait for LinkResolver {
         Ok(data)
     }
 
+    async fn get_block(&self, logger: &Logger, link: &Link) -> Result<Vec<u8>, Error> {
+        if let Some(data) = self.cache.lock().unwrap().get(&link.link) {
```

Comment:
Ah, good catch. The CIDs would refer to the same data, but they are interpreted by the IPFS node differently. So we can't use the same cache. I think it's fine for our use case to avoid caching these results, at least for now.